Raf aligned dom updates (#604)
preactjs/signals: .changeset/ninety-beans-compare.md

---

> "@preact/signals": minor

Author: ```suggestion
"@preact/signals": major
```

---

Defer all DOM updates by an animation frame, this should make it so

> that any used to be synchronous DOM update will be delayed by an

Reviewer: ```suggestion
that any previously synchronous DOM update will be instead delayed by an
```

animation frame. This allows Preact to first perform its own render
cycle and then our direct DOM updates to occur. These will now
be performed in a batched way which is more performant as the browser
is prepared to handle these during the animation frame.

> This does have one way to how Preact based signals are tested, when

Reviewer: Sorry, not quite sure what's meant here but I think there's a typo. "This does impact how..." perhaps?

Author: ```suggestion
This does impact how Preact based signals are tested, when
```

> we perform a signal update we have to wrap it in `act`, in a way
> this was always the case as a signal update that resulted in
> a Preact state update would require it to be wrapped in `act` but

Reviewer: The switch from "we" to "you" is mostly just to emphasize this is a change end users need to take. I think "we" isn't the best word for that as users might think that's an internal change, though I understand your meaning ("we" collectively as users of the lib)

```suggestion
you perform a signal update, you'll need to wrap it in `act`. In a way
this was always the case, as a signal update that resulted in
a Preact state update would require it to be wrapped in `act`, but
```
